### PR TITLE
feat/Task-6: finish working

### PR DESCRIPTION
**Pull request by dmytromykhailiuk**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,28 +1,22 @@
...
-              return <ErrorPage navigateToHome={() => history.push('/')} />  
-            }} />
-            <Redirect to={RouterPaths.ERROR}/>
-          </Switch>
-        </Router>
-        <Footer />
-      </ErrorBoundary>
-    </React.StrictMode>
+    <>  
+      <Routs />
```
Comment (StadnykYura) [inline src/App.tsx]: Routes

File: src/api/movie-api.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import { Categories, SortingOptionsProperties } from "shared/enums";
+import { Movie } from "models/movie.model";
+
+const API_URL = "http://localhost:4000/movies";
+const limit = 12;
+const headers = {
+  "Content-Type": "application/json",
+};
+
+interface GetMoviesOptons {
```
Comment (StadnykYura) [inline src/api/movie-api.ts]: Options

File: src/pages/main/MainPage.tsx [hunk 3]
```diff
@@ -1,130 +1,84 @@
...
-
-  const onEditMovie = useCallback((movie: Movie) => {
-    setModeData({ mode: MainPageModes.EDIT, selectedMovie: movie });
-  }, []);
+const MainPage: React.FC<MainPageProps> = ({ 
+  modalInView, movieInOverview, movieBoard, movieInOverviewLoaded, 
+  movies, isSuccessAlert, selectedMovie, alertMessage, 
+  navigateToMainPage, onCreateMovie, navigateToSearchPage, 
+  setSearchingValue, onEditMovie, onDeleteMovie, onCloseModal, 
+  onCloseWithSaving
```
Comment (StadnykYura) [inline src/pages/main/MainPage.tsx]: make each prop in separate line

File: src/pages/main/MainPageContainer.tsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  const navigateToMainPage = useCallback(() => {
+    history.push(RouterPaths.HOME);
+  }, []);
+
+  const navigateToSearchPage = useCallback(() => {
+    history.push(RouterPaths.SEARCH + "?SearchQuery=");
+  }, []);
+
+  const setSearchingValue = useCallback((searchingValue: string) => {
+    dispatch(MoviesAction.searchMoviesAction({ searchingValue, movieBoard: movieBoard.current }));
```
Comment (StadnykYura) [inline src/pages/main/MainPageContainer.tsx]: i`d rather put a callback as second argument with scrolling than html element

File: src/pages/main/components/MovieBoard/index.ts [hunk 5]
```diff
@@ -1 +1,2 @@
 export { default as MovieBoard } from "./MovieBoard";
+export { default as MovieBoardContainer } from "./MovieBoardContainer";
```
Comment (StadnykYura) [inline src/pages/main/components/MovieBoard/index.ts]: default as MovieBoardComponent is better;
ending Container is used as a hoc frequently, e.g
` export const MovieBoard/Comp = Container(MovieBoard/Component);`

File: src/pages/main/components/MovieItem/MovieItem.tsx [hunk 6]
```diff
@@ -18,6 +20,16 @@ const MovieItem: React.FC<MovieItemProps> = React.memo(({
   onDeleteMovie,
 }) => {
   const [shoudShowMenu, setShoudShowMenuValue] = useState(false);
+  const [hasError, setHasErrorValue] = useState(false);
+
+  useMemo(() => {
```
Comment (StadnykYura) [inline src/pages/main/components/MovieItem/MovieItem.tsx]: why use memo?

File: src/pages/main/components/MovieItemDetails/MovieItemDetails.tsx [hunk 7]
```diff
@@ -1,26 +1,45 @@
...
+  const [hasError, setHasErrorValue] = useState(false);
+
+  useMemo(() => {
+    setHasErrorValue(false);
+  }, [id]);
+
+  const setError = useCallback(() => {
+    setHasErrorValue(true);
+  }, []);
+
```
Comment (StadnykYura) [inline src/pages/main/components/MovieItemDetails/MovieItemDetails.tsx]: duplicated logic try to put inside separate component with image, and why useMemo?

File: src/shared/components/AlertMovieModal/AlertMovieModal.tsx [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  isSuccessAlert: boolean;
+  alertMessage: string;
+  onCloseModal: () => void;
+}
+
+const AlertMovieModal: React.FC<AlertMovieModalProps> = ({ onCloseModal, isSuccessAlert, alertMessage }) => {
+  return (
+    <ModalWrapper onCloseModal={onCloseModal}>
+      <div className={classes['alert-movie-modal__logo']}>
+        { isSuccessAlert ? <CheckboxIcon isLarge={true} /> : "!" }
```
Comment (StadnykYura) [inline src/shared/components/AlertMovieModal/AlertMovieModal.tsx]: `<CheckboxIcon isLarge>`

File: src/shared/components/Loader/Loader.tsx [hunk 9]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import classnames from "classnames";
+import classes from "./Loader.module.scss"
+
+const Loader = () => {
+  return (
+    <div className={classes["loader"]}>
+      <div className={classes["loader__text"]}>Loading...</div>
+      <div className={classes["loader__lines-wrapper"]}>
+        <span className={classnames([classes["loader__line"], classes["loader__line--1"]])}></span>
```
Comment (StadnykYura) [inline src/shared/components/Loader/Loader.tsx:10]: why so featured loader?

File: src/shared/hooks/useShowGoUpButton.ts [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+
+  useEffect(() => {
+    document.addEventListener("scroll", setShowGoUpButton);
+    return () => {
+      document.removeEventListener("scroll", setShowGoUpButton);
+    };
+  }, [showGoUpButton]);
+
+  return showGoUpButton;
+}
```
Comment (StadnykYura) [inline src/shared/hooks/useShowGoUpButton.ts:22]: make addEventlistener to run only once, useRef for that, to hold the referrence to actual handler

Comment (StadnykYura) [reply src/shared/hooks/useShowGoUpButton.ts:22]: you can create even separate custom useEventListener

File: src/store/modals/actions.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { Movie } from "models/movie.model";
+import { createAction } from "@reduxjs/toolkit";
+import { ModalTypes } from "shared/enums";
+
+const prefix = "[Modals]";
+
+interface SetModalInViewActionPeyload {
```
Comment (StadnykYura) [inline src/store/modals/actions.ts]: ..Payload

File: src/store/movies/actions.ts [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+export const loadMovieInOverviewFaildAction = createAction(
+  `${prefix} Load Movie In Overview Faild`
+);
+
+export const clearMovieInOverviewAction = createAction(
+  `${prefix} Clear Movie In Overview`
+);
+
+export const hideLoaderAction = createAction(`${prefix} Hide Loader`);
```
Comment (StadnykYura) [inline src/store/movies/actions.ts:90]: look for this
https://redux-toolkit.js.org/api/createSlice

File: src/store/movies/effects.ts [hunk 13]
```diff
@@ -0,0 +1,169 @@
...
+    yield put(Action.loadMovieInOverviewFaildAction());
+    history.push(RouterPaths.ERROR);
+  }
+}
+
+function* showMoreMoviesWorker() {
+  try {
+    const offset: number = yield select(moviesAmountInStoreSelector);
+    const moviesOptions: MoviesOptions = yield select(moviesOptionsSelector);
+    const searchingValue = getSearchQuery();
```
Comment (StadnykYura) [inline src/store/movies/effects.ts]: i`d rather put his searching value as payload here for this worker

File: src/store/movies/effects.ts [hunk 14]
```diff
@@ -0,0 +1,169 @@
...
+  yield takeLatest(Action.addMovieAction, addMovieWorker);
+  yield takeLatest(Action.editMovieAction, editMovieWorker);
+  yield takeLatest(Action.deleteMovieAction, deleteMovieWorker);
+  yield takeLatest(Action.loadMoreMoviesAction, showMoreMoviesWorker);
+  yield takeLatest(Action.searchMoviesAction, searchMoviesWorker);
+  yield takeLatest(Action.setSortingOptionAction, setSortingOptionWorker);
+  yield takeLatest(Action.setIsDownDirectionValueAction, loadMoviesWorker);
+  yield takeLatest(Action.setSelectedCategoryAction, setSelectedCategoryWorker);
+  yield takeLatest(Action.loadMovieInOverviewAction, loadMovieInOverviewWorker);
+}
```
Comment (StadnykYura) [inline src/store/movies/effects.ts:28]: it is not a best practice to have everything in  in one file

File: src/store/movies/reducer.ts [hunk 15]
```diff
@@ -0,0 +1,203 @@
...
+  moviesAmount: 0,
+  selectedCategory: Categories.ALL,
+  sortingOption: SortingOptionsProperties.RELEASE_DATE,
+  isDownDirection: true,
+  errorMessage: null,
+  moviesLoading: true,
+  moreMoviesLoaded: true,
+};
+
+const moviesReducer = createReducer<MoviesState>(
```
Comment (StadnykYura) [inline src/store/movies/reducer.ts:61]: check for createSlice from documentation

File: src/pages/main/components/MovieItem/MovieItem.tsx (hunk 6 above)
Comment (dmytromykhailiuk) [reply src/pages/main/components/MovieItem/MovieItem.tsx]: to call setHasErrorValue(false) only when movie.id changed

File: src/pages/main/components/MovieItemDetails/MovieItemDetails.tsx (hunk 7 above)
Comment (dmytromykhailiuk) [reply src/pages/main/components/MovieItemDetails/MovieItemDetails.tsx]: Was used useMemo for same reason as above

**Pull request merged**